lib/handlers/events/delivery/goals/FulfillGoalOnRequested.ts

```diff
@@ -164,22 +166,37 @@ export class FulfillGoalOnRequested implements HandleEvent<OnAnyRequestedSdmGoal
                     },
                     implementation,
                     goalInvocation);
-                await reportEndAndClose(result, start, progressLog);
+                const terminatingStates = [
+                    SdmGoalState.canceled,
+                    SdmGoalState.failure,
+                    SdmGoalState.skipped,
+                    SdmGoalState.stopped,
+                    SdmGoalState.success,
```

Comment:
I think the `waiting_for_preapproval` and `waiting_for_approval` states are technically also terminating states in the sense that they come after the `goalExecutor`.

Reply:
`waiting_for_approval` makes sense to me but `waiting_for_preapproval` does not. Wouldn't that state occur _before_ the goal was executed?

Reply:
Oh yes, sorry. You are right. Only `waiting_for_approval` of course.